### DISCUSSION TITLE
Expressing and querying entity relations (projections?)

### DISCUSSION THREAD

**Discussion by MartinKuhne** [ChilliCream/hotchocolate#4232] [Q&A]
So I have two Mongo collections, let's say Author and Book

```
class Author
{
Guid Id {get; set;}
string Name {get; set;}
ICollection<Book> Books {get; set;}
}
class Book
{
Guid Id {get; set;}
Guid AuthorId {get; set;}
string Name {get; set;}
}
```

and I pretty much have the boilerplate query
```
class Query
{
        [UseSorting]
        [UseFiltering]
        [UseProjection]
        public IExecutable<Author> GetAuthors([Service] IMongoCollection<Author> collection)
        {
            return collection.AsExecutable();
        }
}
```

and the same for Book

But when I say

```
{
authors
  {
    id
    name
    books
    {
      id
      name
    }
  }
}
```

books is null

How is the linkage expressed for the relation? How do I make it get SELECT * FROM BOOKS WHERE Id = AuthorId? I feel like I am missing some fundamental concept here. Like in entity framework, there was an implicit contract where as long as you had Id and AuthorId properties, it would recognize the foreign key relationship.

TIA
Martin

**Comment by farhadnowzari** [comment]
Hi @MartinKuhne ,
I think you have lazy loading disabled on ef core, if so you can Include the relation when you are building the query for HotChocolate.
This is how I handle this.
it something like this

```csharp
public IQueryable<Author> GetAuthors([Service] ApplicationDbContext dbContext) => dbContext.Authors.Include(x => x.Books);
```

**Comment by ghost** [comment]
This question does not relate to EF Core, as it clearly states two **Mongo** collections.

MongoDB, as a NoSQL database, does not understand/support/enforce foreign key relationships automatically.

Even if the EF Core suggestion of using ".Include" was possible, it would be a bad idea, leading to significant over-fetching of data. Do you want to retrieve the list of books every time you fetch an author?

The correct way to handle this in HotChocolate is to specify the resolver needed to retrieve the Books field inside the Author object type.

Exactly how you do this will depend on the style of code you are following, but essentially will involve writing the MongoDB query to retrieve all Books with a certain AuthorId. (You should also build an index on the Books collection by AuthorId, otherwise this will involve a sequential search of all the Books in the collection.)
